components/server-side-render/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies.
+ */
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: null,
+		};
+	}
+
+	componentDidMount() {
+		this.fetch( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps, this.props ) ) {
+			this.fetch( nextProps );
+		}
+	}
+
+	fetch( props ) {
+		this.setState( { response: null } );
+		const { block } = props;
+		const attributes = Object.assign( {}, props );
+
+		// Delete 'block' from attributes, only registered block attributes are allowed.
+		delete attributes.block;
+
+		let apiURL = this.getQueryUrlFromObject( {
+			attributes: attributes,
+		} );
+
+		apiURL = '/gutenberg/v1/block-renderer/' + block + '?' + apiURL;
+
+		return wp.apiRequest( { path: apiURL } ).then( response => {
+			if ( response && response.rendered ) {
+				this.setState( { response: response.rendered } );
+			}
+		} );
+	}
+
+	getQueryUrlFromObject( obj, prefix ) {
+		const str = [];
+		let param;
+		for ( param in obj ) {
+			if ( obj.hasOwnProperty( param ) ) {
+				const key = prefix ? prefix + '[' + param + ']' : param,
+					value = obj[ param ];
+				str.push(
+					( value !== null && 'object' === typeof value ) ?
+						this.getQueryUrlFromObject( value, key ) :
+						encodeURIComponent( key ) + '=' + encodeURIComponent( value )
+				);
+			}
+		}
+		return str.join( '&' );
+	}
```

Comment:
1. The general body of this method could be simplified as:

```js
import { map } from 'lodash';
return map( obj, ( paramValue, paramName ) => {
  const key = prefix ? …;
  return isObject( paramValue ) ? this.getQ… : encodeURI…;
} ).join( '&' );
```

2. A more proper solution might be to review `@wordpress/url`'s `addQueryArgs` and its choice of `querystring`, if nested attributes are deemed necessary.

Reply:
Thank you, replaced the method in https://github.com/WordPress/gutenberg/pull/5602/commits/e0fec83b254732ede34a69e54eaf531b86360528. 
Note that I checked [querystring](https://nodejs.org/api/querystring.html) and also [query-string](https://www.npmjs.com/package/query-string), these don't seem to support nested objects.